Bottom: dbfb284f0129949410cd41702e47a57f9e225610
Top:    8f22843b8571fbe15ffbc5112aafe4c1a1ab952e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:58:40 +0000

hid/gcode: Fix memory leak in error path

(Caught with cppcheck)


---

diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 40a3b36..17bbae1 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -558,6 +558,8 @@ gcode_do_export (HID_Attr_Val * options)
           if (!gcode_f2)
             {
               perror (filename);
+              free (filename);
+              bm_free (bm);
               return;
             }
           fprintf (gcode_f2, "(Created by G-code exporter)\n");
